src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppellantNameForDisplayFormatterTest.java

```diff
@@ -101,25 +106,86 @@ public void handling_should_throw_if_cannot_actually_handle() {
     public void it_can_handle_callback() {
 
         for (Event event : Event.values()) {
-
             when(callback.getEvent()).thenReturn(event);
 
             for (PreSubmitCallbackStage callbackStage : PreSubmitCallbackStage.values()) {
+                when(callback.getCaseDetails()).thenReturn(caseDetails);
+                when(caseDetails.getCaseData()).thenReturn(asylumCase);
+                when(asylumCase.read(JOURNEY_TYPE)).thenReturn(Optional.empty());
 
                 boolean canHandle = appellantNameForDisplayFormatter.canHandle(callbackStage, callback);
 
                 if (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT) {
-
-                    assertTrue(canHandle);
+                    assertTrue("Can handle event " + event, canHandle);
                 } else {
-                    assertFalse(canHandle);
+                    assertFalse("Cannot handle event " + event, canHandle);
                 }
             }
 
             reset(callback);
         }
     }
 
+    @Test
+    public void cannot_handle_Aip_start_event() {
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        when(asylumCase.read(JOURNEY_TYPE)).thenReturn(Optional.of(JourneyType.AIP));
+        when(callback.getEvent()).thenReturn(Event.START_APPEAL);
+
+        boolean canHandle = appellantNameForDisplayFormatter.canHandle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
+
+        assertThat(canHandle, is(false));
+    }
+
+    @Test
+    public void cannot_handle_Aip_edit_event() {
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        when(asylumCase.read(JOURNEY_TYPE)).thenReturn(Optional.of(JourneyType.AIP));
+        when(callback.getEvent()).thenReturn(Event.EDIT_APPEAL);
+
+        boolean canHandle = appellantNameForDisplayFormatter.canHandle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
+
+        assertThat(canHandle, is(false));
+    }
+
+    @Test
+    public void can_handle_Aip_other_event() {
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        when(asylumCase.read(JOURNEY_TYPE)).thenReturn(Optional.of(JourneyType.AIP));
+        when(callback.getEvent()).thenReturn(Event.BUILD_CASE);
+
+        boolean canHandle = appellantNameForDisplayFormatter.canHandle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
+
+        assertThat(canHandle, is(true));
```

Comment:
We've been using AssertJ for fluent assertions.  Nothing wrong with hamcrest but rather not mix too many different styles if possible.